conf/stats.conf.example

```diff
@@ -2,96 +2,105 @@
 type=mysql_query
 statsd_type=gauge
 statsd_ns=source.packetfence.devices.registered_per_role
-mysql_query=select name, count(distinct node.mac) from node_category right join node on node.category_id=node_category.category_id where node.status='reg' group by node_category.category_id;
+mysql_query=select name, count(distinct node.mac) from node_category left join node on node.category_id=node_category.category_id and node.status='reg' group by node_category.category_id
 interval=60s
+randomize=true
 
 [metric 'total number of unregistered devices']
 type=mysql_query
 statsd_type=gauge
 statsd_ns=source.packetfence.devices.unregistered
 mysql_query=select count(distinct mac) from node where status='unreg'
 interval=60s
+randomize=true
 
 [metric 'total online devices']
 type=mysql_query
 statsd_type=gauge
 statsd_ns=source.packetfence.devices.online
 mysql_query=select count(distinct callingstationid) from radacct where acctstoptime is null
 interval=60s
+randomize=true
 
 [metric 'total offline devices']
 type=mysql_query
 statsd_type=gauge
 statsd_ns=source.packetfence.devices.offline
 mysql_query=select count(distinct callingstationid) from radacct where acctstoptime <= NOW()
 interval=60s
+randomize=true
 
 [metric 'number of new registered devices during the past hour']
 type=mysql_query
 statsd_type=gauge
 statsd_ns=source.packetfence.devices.registered_last_hour
 mysql_query=select count(distinct mac) from node where status='reg' and regdate >= NOW() - INTERVAL 1 HOUR
-interval=60s
+interval=1s
 
 [metric 'number of new registered devices during the past day']
 type=mysql_query
 statsd_type=gauge
 statsd_ns=source.packetfence.devices.registered_last_day
 mysql_query=select count(distinct mac) from node where status='reg' and regdate >= NOW() - INTERVAL 1 DAY
-interval=60s
+interval=24s
 
 [metric 'number of new registered devices during the past week']
 type=mysql_query
 statsd_type=gauge
 statsd_ns=source.packetfence.devices.registered_last_week
 mysql_query=select count(distinct mac) from node where status='reg' and regdate >= NOW() - INTERVAL 1 WEEK
-interval=60s
+interval=168s
 
 [metric 'number of new registered devices during the past month']
 type=mysql_query
 statsd_type=gauge
 statsd_ns=source.packetfence.devices.registered_last_month
 mysql_query=select count(distinct mac) from node where status='reg' and regdate >= NOW() - INTERVAL 1 MONTH
-interval=60s
+interval=720s
 
 [metric 'number of new registered devices during the past year']
 type=mysql_query
 statsd_type=gauge
 statsd_ns=source.packetfence.devices.registered_last_year
 mysql_query=select count(distinct mac) from node where status='reg' and regdate >= NOW() - INTERVAL 1 YEAR
-interval=60s
+interval=8760s
```

Comment:
Can't we make it evaluated every hour ?

Otherwise the user has to wait a very long time for data to refresh, at that point its not even worth putting into a graph

Reply:
@julsemaan this is the point of the smaller timeframe graphs (hour/day/week/month). Adding more samples in the "year" graph will reduce the minimum x-axis on its graph, where I think it's useful to have an entire year in the year graph, an entire month in the month graph, etc.

We want 3600 samples over 1 year.

24h = (60 * 60 * 24 * 365) = 31536000 seconds

31536000 seconds / 3600 samples = 8760 seconds/sample

Reply:
This logic makes sense, only thing is that I think we should cap this to sampling each 24 hours since some users have a tendency to restart services often so they'd be missing out on the stats

Reply:
> they'd be missing out on the stats

the longer time-based charts omit `randomize=true` in the config. This forces them to run immediately after restart, thus avoiding any gaps produced by frequent restarts.